src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -428,32 +435,37 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
           const created = createFiberFromElement(newChild, priority);
           created.ref = coerceRef(null, newChild);
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
 
         case REACT_COROUTINE_TYPE: {
           const created = createFiberFromCoroutine(newChild, priority);
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
 
         case REACT_YIELD_TYPE: {
           const created = createFiberFromYield(newChild, priority);
           created.type = newChild.value;
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
 
         case REACT_PORTAL_TYPE: {
           const created = createFiberFromPortal(newChild, priority);
           created.return = returnFiber;
+          created.contextTag = returnFiber.contextTag;
           return created;
         }
       }
 
       if (isArray(newChild) || getIteratorFn(newChild)) {
         const created = createFiberFromFragment(newChild, priority);
         created.return = returnFiber;
+        created.contextTag = returnFiber.contextTag;
```

Comment:
@flarnie The `return` pointer is not just a parent pointer because there can be more than one parent. It's also the pointer to the fiber to work on next so that we don't have to also keep a separate stack. Therefore we need to update it when we traverse down the tree.

We don't need to do that with the `contextTag`. So I'd be careful about coupling these into a single concept. We might want to decouple the return pointer to just use a stack instead and maybe even drop the parent pointer. That's why we try to err on the side of less abstraction because it can be subtle.